File: BiteClubAPI/src/db/schema/receiptDetails.ts
import { z, ZodSchema } from "zod";

export const receiptDetails = z.object({
    items: z.array(
        z.object({
            name: z.string().nonempty(),
            unit_price: z.number().min(0),
            quantity: z.number().min(0),
        })
    ),
    subtotal: z.number().min(0),
    tax: z.number().min(0),
    total: z.number().min(0),
});

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/DTO/ReceiptDetails.kt
package org.biterunr_studios.biterunr.DTO

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ReceiptDetails(
    @SerialName("items")
    val items: List<Item>,
    @SerialName("subtotal")
    val subtotal: Double,
    @SerialName("tax")
    val tax: Double,
    @SerialName("total")
    val total: Double
) {
    @Serializable
    data class Item(
        @SerialName("name")
        val name: String,
        @SerialName("unit_price")
        val unitPrice: Double,
        @SerialName("quantity")
        val quantity: Double
    )
}
